nodes/node_registry.go

```diff
@@ -63,31 +63,106 @@ func (r *NodeRegistry) GetRegisteredNodeKindNames() []string {
 	return result
 }
 
+func (r *NodeRegistry) GetGenerateNodeAttributes() map[string]*GenerateNodeAttributes {
+	result := map[string]*GenerateNodeAttributes{}
+	for k, v := range r.nodeIndex {
+		result[k] = v.GetGenerateAttributes()
+	}
+	return result
+}
+
 func (r *NodeRegistry) Kind(kind string) *NodeRegistryEntry {
 	return r.nodeIndex[kind]
 }
 
 type NodeRegistryEntry struct {
 	nodeKindNames []string
 	initFunction  Initializer
-	credentials   *Credentials
+	attributes    *NodeRegistryEntryAttributes
+}
+
+func (nre *NodeRegistryEntry) GetCredentials() *Credentials {
+	if nre.attributes == nil {
+		return nil
+	}
+	return nre.attributes.credentials
+}
+
+func (nre *NodeRegistryEntry) GetGenerateAttributes() *GenerateNodeAttributes {
+	if nre.attributes == nil {
+		return nil
+	}
+	return nre.attributes.generateAttributes
 }
 
 // Credentials returns entry's credentials.
-func (e *NodeRegistryEntry) Credentials() *Credentials {
+// might return nil if no default credentials present.
+func (e *NodeRegistryEntryAttributes) Credentials() *Credentials {
 	if e == nil {
 		return nil
 	}
 
 	return e.credentials
 }
 
-func newRegistryEntry(nodeKindNames []string, initFunction Initializer, credentials *Credentials) *NodeRegistryEntry {
+func newRegistryEntry(nodeKindNames []string, initFunction Initializer,
+	attributes *NodeRegistryEntryAttributes,
+) *NodeRegistryEntry {
 	return &NodeRegistryEntry{
 		nodeKindNames: nodeKindNames,
 		initFunction:  initFunction,
-		credentials:   credentials,
+		attributes:    attributes,
+	}
+}
+
+type NodeRegistryEntryAttributes struct {
+	credentials        *Credentials
+	generateAttributes *GenerateNodeAttributes
+}
+
+func NewNodeRegistryEntryAttributes(c *Credentials, ga *GenerateNodeAttributes) *NodeRegistryEntryAttributes {
+	// set default value for GenerateNodeAttributes
+	if ga == nil {
+		ga = NewGenerateNodeAttributes(false, "")
+	}
+	return &NodeRegistryEntryAttributes{
+		credentials:        c,
+		generateAttributes: ga,
+	}
+}
+
+func (nrea *NodeRegistryEntryAttributes) GetCredentials() *Credentials {
+	return nrea.credentials
+}
+
+func (nrea *NodeRegistryEntryAttributes) GetGenerateAttributes() *GenerateNodeAttributes {
+	return nrea.generateAttributes
+}
+
+type GenerateNodeAttributes struct {
+	generateable    bool
```

Comment:
likely we don't need the `generatable` flag at all. All nodes should be generatable, I think some nodes were not generatable originally  because we just did not add them manually to the generage cmd